src/foam/core/Reaction.js

```diff
@@ -0,0 +1,72 @@
+foam.CLASS({
+  package: 'foam.core',
+  name: 'Reaction',
+  properties: [
+    {
+      name: 'name',
+      expression: function(target, topic, listener) {
+        return 'reaction_' + target +  '$$' + topic + '$$' + listener;
+      }
+    },
+    {
+      class: 'String',
+      name: 'target'
+    },
+    {
+      class: 'StringArray',
+      name: 'topic'
+    },
+    {
+      name: 'listener'
+    },
+  ],
+  methods: [
+    function initObject(obj) {
+      var path = this.target.split('.');
```

Comment:
Would be more efficient if target were a StringArray with an adapt method to do the split if you just wanted to provide a string rather than an array. Would be faster to init an create fewer objects.